core/src/main/java/org/jruby/runtime/load/LibrarySearcher.java

```diff
@@ -133,6 +132,20 @@ private FoundLibrary findResourceLibrary(String baseName, String suffix) {
           return findFileResource(baseName, suffix);
         }
 
+        // ruby does not load a relative path unless the current working directory is in $LOAD_PATH
+        if (baseName.startsWith("./") || loadService.loadPath.contains(".")) {
```

Comment:
Is there a test that requires to skip ./ check if . is in load path? This seems rather strange to me, since I thought MRI treats ./ as a pretty special file index. Also, MRI (2.1.2 at least) seems to bypass LOAD_PATH, even if it contains .:

$ ruby -Ilib:. -e 'require "./foo"'
local

(the lib contains another foo.rb that would have printed 'in lib')